src/control/destination/background/mod.rs

```diff
@@ -484,3 +484,36 @@ where
         });
     }
 }
+
+// ===== impl DestinationServiceQuery =====
+
+impl<T: HttpService<ResponseBody = RecvBody>> DestinationServiceQuery<T> {
+
+    pub fn is_active(&self) -> bool {
+        match self {
+            DestinationServiceQuery::Active(_) => true,
+            _ => false,
+        }
+    }
+
+    pub fn wants_query(&self) -> bool {
```

Comment:
I think this would be slightly clearer if it referred to capacity explicitly.  "wants query" makes me wonder why it wants a query, or has any desires at all.  `needs_query_capacity()` ?